http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala

```diff
@@ -1098,57 +1083,42 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
                                               wsmManagedIdentityOpt: Option[WsmManagedAzureIdentity]
   )(implicit ev: Ask[F, AppContext]): F[WsmControlledKubernetesNamespaceResource] =
     for {
-      wsmNamespaces <- retrieveWsmNamespace(resourceApi, namespacePrefix, workspaceId.value)
-      namespace <- wsmNamespaces.length match {
-        case 1 =>
+      wsmNamespace <- retrieveWsmNamespace(resourceApi, namespacePrefix, workspaceId.value)
+      namespace <- wsmNamespace match {
+        case Some(ns) =>
           logger.info(
-            s"Namespace found in WSM for app ${app.appName}, using previously created namespace: ${wsmNamespaces.head.name}"
+            s"Namespace found in WSM for app ${app.appName}, using previously created namespace: ${ns.name}"
           )
-          F.pure(wsmNamespaces.head)
-        case 0 =>
+          F.pure(ns)
+        case None =>
           createWsmKubernetesNamespaceResource(
             app,
             workspaceId,
             namespacePrefix,
             wsmDatabases.map(_.wsmDatabaseName),
             wsmManagedIdentityOpt.map(_.wsmResourceName)
           )
-        case _ =>
-          F.raiseError(
-            AppCreationException(
-              s"App ${app.appName} has multiple namespaces $wsmNamespaces. Only one namespace per app allowed"
-            )
-          )
       }
     } yield namespace
 
   private[util] def retrieveWsmNamespace(resourceApi: ResourceApi,
                                          namespacePrefix: String,
                                          workspaceId: UUID
-  ): F[List[WsmControlledKubernetesNamespaceResource]] = {
-
-    val wsmNamespaces = F.blocking(
-      resourceApi
-        .enumerateResources(workspaceId, 0, 100, ResourceType.AZURE_KUBERNETES_NAMESPACE, StewardshipType.CONTROLLED)
-        .getResources
-        .asScala
-        .toList
-    )
-    wsmNamespaces.map { namespaces =>
-      namespaces
-        .filter(ns =>
-          ns.getResourceAttributes.getAzureKubernetesNamespace.getKubernetesNamespace.startsWith(namespacePrefix)
-        )
-        .map(filtered_ns =>
-          WsmControlledKubernetesNamespaceResource(
-            NamespaceName(filtered_ns.getResourceAttributes.getAzureKubernetesNamespace.getKubernetesNamespace),
-            WsmControlledResourceId(filtered_ns.getMetadata.getResourceId),
-            ServiceAccountName(
-              filtered_ns.getResourceAttributes.getAzureKubernetesNamespace.getKubernetesServiceAccount
-            )
+  ): F[Option[WsmControlledKubernetesNamespaceResource]] = {
+    // The full namespace name will be {namespacePrefix}-{workspaceId},
```

Comment:
Is this a safe assumption to make? Who decides what the namespace name will be?

Reply:
I think this logic is baked into WSM. It would be nice to at least link the logic where this is defined in WSM. But there is no real contract so if the logic changes this would break.

Reply:
It is indeed baked into WSM, which I agree is a bit scary. 
In principal, this is exactly the sort of assumption contract tests would be useful to test. 

I'll add an addition to the comment to spell out where it's happening.

Reply:
Is namespacePrefix constrained in length? I'm wondering if we need to worry about possible truncation (in WSM).

Reply:
That is a good question. Leo is not creating the namespacePrefix but getting it from WSM via the `AzureKubernetesNamespaceCreationParameters` so I am not sure if there are any constraints there.

Reply:
`namespacePrefix` looks to be limited in WSM to 24 characters (`service/build/openapi/src/main/java/bio/terra/workspace/generated/model/ApiAzureKubernetesNamespaceCreationParameters.java`)

I _think_ it's typically the app name: at least that's what it looked like from what I saw in the test data, as well as during some debugging.